runtime/near-vm-logic/src/logic.rs

```diff
@@ -2430,3 +2430,18 @@ pub struct VMOutcome {
     pub used_gas: Gas,
     pub logs: Vec<String>,
 }
+
+impl std::fmt::Display for VMOutcome {
```

Comment:
I would say it is an abuse of `Display`. The information is useful in logging. Let's consider IPv4 address as an example; Display makes perfect sense there since its internal implementation is a 4-byte integer which can be displayed to humans as `127.0.0.1`. Here, I argue that "VMOutcome: balance ..." is not a display of the VMOutcome, there is no meaningful Display for this structure, so I would avoid implementing it. Instead, I would implement a simple `.format_log_message()` / `.display_log_message()` / `.to_log_message()` / `.fmt_log_message()` (naming is hard, you know). We also may just as well replace the derived `Debug` implementation with this custom one.

Reply:
Yeah, agree that there's no meaningful Display possible here. Hand-writing debug is what I'd do personally. Here's an example of compressed debug from rust-analyzer: 

https://github.com/rust-analyzer/rust-analyzer/blob/fc709c8b210c551c97ca3b7c7daeaabc15cd80aa/crates/project_model/src/workspace.rs#L53-L79

If we want do be fancy (we don't), we can even write it such that `{:?}` prints compressed representation and `{:#?}` prints the full one. 

I would include the length of `Value` into the debug repr though.